module/VuFind/src/VuFind/ILS/Driver/Alma.php

```diff
@@ -225,6 +232,29 @@ protected function getAvailabilityFromItem($item)
         return (string)$item->item_data->base_status === '1';
     }
 
+    /**
+     * Get the item limit from Alma.ini config file. It the config is not set the
+     * default value is 10.
+     *
+     * @return int
+     */
+    public function getItemLimit()
```

Comment:
This value will already be sent back by `getConfig('Holds')` calls. I wonder if this is a more generic, existing way for the calling code to gain access to this value without having to change the driver API (though of course that would require the defaulting/normalization logic to be moved to a different place or places). Did you consider this possibility? This might be a good way to do feature degradation -- the other drivers won't return an itemLimit value in the holds configuration, so VuFind will know it can't do pagination with those drivers. You could perhaps include a special case in the getConfig implementation in the Alma driver to force a default value for the setting if pagination is required here (though admittedly, that gets ugly).

I'm considering revising the ILS driver spec in future to return more tightly structured responses (using classes extending ArrayObject instead of free-form arrays) which may allow us to clean up some of this stuff... but until we have time to do a more comprehensive spec review/revision (discussed in more detail at [VUFIND-1280](https://vufind.org/jira/browse/VUFIND-1280)), I'm trying to make as few disruptive changes to it as possible.

All that being said, if you still feel a new method is absolutely necessary, I'm not closed to the idea -- but it should have a more specific name than `getItemLimit`; at very least, `getHoldsItemLimit`.

Reply:
Thank you for pointing this out. I will try using `getConfig('Holds')`

Reply:
I use `getConfig('Holds')` for getting the "itemLimit" config from Alma.ini now. I call it in "RecordTab\HoldingsILS" because the value is needed there for calculating the "offset", for adding it to the new `$options` array (more information on that below) and for the paginators `setItemCountPerPage()`. The "itemLimit" will be null when not set so that VuFin can find out if paginagion is supported or not. If not, the paginator won't be echoed in "RecordTab\holdingsils.phtml". As you suggested I also included a special case in the `getConfig()` implementation in the Alma driver for the case that the value for "itemLimit" is not set or is empty. I think it's not that ugly (just a bit :-) ).